i6_models/assemblies/lstm/lstm_v1.py

```diff
@@ -0,0 +1,87 @@
+__all__ = [
+    "LstmEncoderV1Config",
+    "LstmEncoderV1",
+]
+
+from dataclasses import dataclass
+import torch
+from torch import nn
+from typing import Any, Dict, Optional, Tuple, Union
+
+from i6_models.config import ModelConfiguration
+from i6_models.parts.lstm import LstmBlockV1Config, LstmBlockV1
+
+
+@dataclass
+class LstmEncoderV1Config(ModelConfiguration):
+    """
+    :param init_args: used to initialize parameters of modules, example:
+        ```
+        {
+            "init_args_w": {"func": "normal", "arg": {"mean": 0.0, "std": 0.1}},
+            "init_args_b": {"func": "normal", "arg": {"mean": 0.0, "std": 0.1}},
+        }
+        ```
+    """
+
+    input_dim: int
+    embed_dim: int
+    embed_dropout: float
+    lstm_layers_cfg: LstmBlockV1Config
+    lstm_dropout: float
+    init_args: Optional[Dict[str, Any]] = None
+
+    @classmethod
+    def from_dict(cls, model_cfg_dict: Dict[str, Any]):
+        model_cfg_dict = model_cfg_dict.copy()
+        model_cfg_dict["lstm_layers_cfg"] = LstmBlockV1Config.from_dict(model_cfg_dict["lstm_layers_cfg"])
+        return cls(**model_cfg_dict)
+
+
+class LstmEncoderV1(nn.Module):
+    def __init__(self, model_cfg: Union[LstmEncoderV1Config, Dict[str, Any]], **kwargs):
+        """
+        Model definition of LSTM encoder. Contains embedding layer followed by single lstm stack, dropout after both.
+        Padding sequence in forward call.
+
+        :param model_cfg: holds model configuration as dataclass or dict instance.
+        :param kwargs:
+        """
+        super().__init__()
+
+        self.cfg = LstmEncoderV1Config.from_dict(model_cfg) if isinstance(model_cfg, dict) else model_cfg
+
+        self.embedding = nn.Embedding(self.cfg.input_dim, self.cfg.embed_dim)
+        self.embed_dropout = nn.Dropout(self.cfg.embed_dropout)
+
+        self.lstm_block = LstmBlockV1(self.cfg.lstm_layers_cfg)
+        self.lstm_dropout = nn.Dropout(self.cfg.lstm_dropout)
+
+        if self.cfg.init_args is not None:
+            self._param_init(**self.cfg.init_args)
+
+    def _param_init(self, init_args_w=None, init_args_b=None):
+        for m in self.modules():
+            for name, param in m.named_parameters():
```

Comment:
I think I would be a bit more cautious here. This would fail with `KeyError` if given something that does not have `func` and `arg` in it. Maybe assert this beforehand? should we maybe even enforce it in the config above via post init?

Reply:
Added a post init to handle this.